implentio/ui-toolkit
# added strict type guard for onRemove prop

# Pull Request Template

## Description
onRemove is a prop which is only required and useful when addView = false. However, it can currently be input when addView=true. This PR changes the typing so that onRemove cannot be passed when addView=true

## Link to JIRA ticket
[Include the link here.](https://implentio.atlassian.net/jira/software/projects/IUT/boards/10?selectedIssue=IUT-35)

## Testing Instructions
Outline the steps required to test your changes effectively (Replace the following to what makes sense for your PR):
- Fetch and checkout my branch `git fetch && git checkout <fix/IUT-35/Update-FilterChip-w-type-guard>`
- Go to src/components/FilterChip/FilterChip.tsx
- on the bottom of the page, copy and paste: `<FilterChip addView={false} value="value" filter="filter: " onRemove={()=> console.log("test")} />`
- check if typescript throws type errors when you add and remove props and when you toggle addView to true/false

## Checklist:
Check what is applicable by putting an 'x' in between the []:
